player/public/coreplugins/org.ekstep.pdfrenderer-1.0/renderer/plugin.js

```diff
@@ -205,6 +209,40 @@ org.ekstep.contentrenderer.baseLauncher.extend({
         context.showPDF(path, context.manifest);
     },
 
+    addDownloadButton: function(path, pdfSearchContainer){
+        if(!path.length) return false;
+        var instance = this;
+        var downloadBtn = document.createElement("img");
+        downloadBtn.id = "download-btn"; 
+        downloadBtn.src = EkstepRendererAPI.resolvePluginResource(instance.manifest.id, instance.manifest.ver, "renderer/assets/mdpi.png");
+        downloadBtn.className = "pdf-download-btn";
+        downloadBtn.onclick = function(){
+            var link = document.createElement('a');
```

Comment:
Why do you need link element? you just need a path to download.

Reply:
Yes, this is needed for auto download the pdf file - when a user clicks on the download button.

if we use `window.open(url, 'dowload')` and browser supports the pdf then this pdf file will open in a new tab and user again need to click on the download icon to download this same pdf. To avoid this - we need temporary link element